repo/packages/J/jenkins/17/config.json

```diff
@@ -0,0 +1,133 @@
+{
+    "type": "object",
+    "properties": {
+        "service": {
+            "description": "Configuration properties for the Jenkins service for DC/OS.",
+            "type": "object",
+            "properties": {
+                "name": {
+                    "description": "The name of the service to display in the DC/OS dashboard.",
+                    "type": "string",
+                    "default": "jenkins"
+                },
+                "cpus": {
+                    "description": "CPU shares to allocate to each Jenkins master.",
+                    "type": "number",
+                    "default": 1.0,
+                    "minimum": 0.1
+                },
+                "mem": {
+                    "description": "Memory (in MB) to allocate to each Jenkins master.",
+                    "type": "number",
+                    "default": 2048.0,
+                    "minimum": 2048.0
+                }
+            },
+            "required": [
+                "name",
+                "cpus",
+                "mem"
+            ]
+        },
+        "storage": {
+            "description": "Storage-related configuration properties for Jenkins on DC/OS.",
+            "type": "object",
+            "properties": {
+                "host-volume": {
+                    "description": "The location of a volume on the host to be used for persisting Jenkins configuration and build data. The final location will be derived from this value plus the name set in `name` (e.g. `/mnt/host_volume/jenkins`). Note that this path must be the same on all DC/OS agents.",
+                    "type": "string",
+                    "default": "/tmp"
```

Comment:
Persistent volumes do not work for us since they would be deleted by Marathon upon an uninstall of the Jenkins app (which would destroy the Jenkins data). There is currently no option to avoid this. Since this is our upgrade path for Jenkins (uninstall Jenkins and reinstall) - there's no way to do this cleanly short of writing another framework - which is something we are trying to avoid (since then we'd have two frameworks - a framework to launch the Jenkins master which contains another framework to launch Jenkins agents).

Overriding the default `host-volume` config value is extensively documented in the service documentation for Jenkins and is currently the best option we have to protect our users' data.